### PR TITLE
feat: add `go-task` spec

### PR DESCRIPTION
**Pull request by yoavbls**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
feature, add spec for [go-task](taskfile.dev)

**What is the current behavior? (You can also link to an open issue here)**
Currently, fig shows autocompletion only for `taskwarrior` binary when writing `task ...`

**What is the new behavior (if this is a feature change)?**
fig will show autocompletion for `go-task` if it is the binary that runs on `task`, or `taskwarrior` if it isn't

CleanShot 2022-03-19 at 19 18 36@2x

**Additional info:**
Unfortunately, autocompletion appears also when writing `go-task ...` or `taskwarrior ...`
I would like to know about a way to disable it, I can't use `loadSpec` because it doesn't work for root commands (#223)

### CODE REVIEW

**Comment by fedeci** [comment]
> **Additional info:** Unfortunately, autocompletion appears also when writing `go-task ...` or `taskwarrior ...` I would like to know about a way to disable it

I still have to review this PR but you can nest `taskwarrior.ts` and `go-task.ts` under a directory e.g. `task/taskwarrior.ts`, `task/go-task.ts`

**Comment by davidalpert** [comment]
@fedeci or @mschrage could we please review and merge this PR?

I have run into this exact issue recently and popped over to investigate/fix it only to find that @yoavbls has already fixed it! 😍 

I just checked out this branch locally, ran the autocomplete specs in dev mode, and verified that this PR does in fact replace the taskwarrior autocompletions with the tasks from my taskfile. 🎉

**Comment by fedeci** [comment]
I am giving a look!

**Comment by fedeci** [comment]
recheck

**Comment by yoavbls** [comment]
@fedeci where do I accept it?
Do you want me to resolve the conflicts? Is someone going to review it?

**Comment by fedeci** [comment]
@yoavbls I am reviewing right now, don't worry about conflicts, I am fixing those!
Sorry for late response!

**Comment by yoavbls** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/task/go-task.ts [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+      context = ` ${tokens[contextValueIndex - 1]} ${
+        tokens[contextValueIndex]
+      }`;
+    }
+
+    const tasksListText = await exec(`task${context} --list`);
+
+    return tasksListText
+      .split("\n* ")
+      .filter((_, i) => i > 0) // The first line is not a task
```
Comment (fedeci) [inline src/task/go-task.ts:33]:
```suggestion
      .slice(1) // The first line is not a task
```

File: src/task/go-task.ts [hunk 2]
```diff
@@ -0,0 +1,163 @@
+const TASKS_PRIORITY = 80;
+const TASKFILE_FLAGS = ["-t", "--taskfile"];
+const DIRECTORY_FLAGS = ["-d", "--dir"];
```
Comment (fedeci) [inline src/task/go-task.ts:3]:
```suggestion
const DIRECTORY_FLAGS = ["-d", "--dir"];
const taskDirectoryFlags = new Set([...TASKFILE_FLAGS, ...DIRECTORY_FLAGS]);
```

File: src/task/go-task.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    }
+
+    let context = "";
+    const contextValueIndex =
+      tokens.length -
+      tokens
+        .slice()
+        .reverse()
+        .findIndex((token) =>
+          [...TASKFILE_FLAGS, ...DIRECTORY_FLAGS].includes(token)
```
Comment (fedeci) [inline src/task/go-task.ts:19]:
```suggestion
          taskDirectoryFlags.has(token)
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks, I left a couple perf improvements but it looks great!

**Review by fedeci — CHANGES_REQUESTED** [review]

**Review by fedeci — APPROVED** [review]

**Pull request merged**